### DISCUSSION TITLE
Does lhotse support multiple features?

### DISCUSSION THREAD

**Discussion by g6w9a** [lhotse-speech/lhotse#1397] [Q&A] [answered]
I read the documentation and the codebase. I think a cut can only have one features. I was looking for something like this:
```python
cuts = load_manifest_lazy("path/to/manifest")
cut = cuts[0]

mfcc_features = cut.load_features(type="mfcc")
fbank_features = cut.load_features(type="kaldi-fbank")
```
`cut.features` can only hold information about one feature. 

Is there a way to work with multiple features?

**Comment by g6w9a** [comment]
If this is not possible, I would like to work on a PR about this. Does this also make sense for many use cases?

**Reply by pzelasko** [reply, marked as answer]
It's possible, but poorly documented as of yet. You want to check the `FeaturesWriter.store_array` used like this:

```python
with NumpyFilesWriter(path) as w:  # works with other writers too
    cut.my_custom_features = w.store_array(key, array, ...)
```

**Comment by g6w9a** [comment]
Thanks. By following @pzelasko 's suggestion I managed to save custom features with hd5 writer. I end up writing GenericFeatures class to easily load and manipulate those features later on.
